packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -7,98 +7,116 @@ pragma solidity 0.8.10;
+    /// @param semanticVersion The semantic version number.
+    event VersionCreated(uint256 versionId, uint16[3] semanticVersion);
+
+    /// @notice Initializes the contract by
+    /// - registering the [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID
+    /// - initializing the permission manager
+    /// - setting the next version index to 1 and
+    /// - giving the `CREATE_VERSION_PERMISSION_ID` permission to the initial owner.
+    /// @dev This method is required to support [ERC-1822](https://eips.ethereum.org/EIPS/eip-1822).
     function initialize(address initialOwner) external initializer {
         _registerStandard(type(IPluginRepo).interfaceId);
-        __ACL_init(initialOwner);
+        __PermissionManager_init(initialOwner);
 
         nextVersionIndex = 1;
 
-        // set roles.
-        _grant(address(this), initialOwner, CREATE_VERSION_ROLE);
+        // set permissionIDs.
+        _grant(address(this), initialOwner, CREATE_VERSION_PERMISSION_ID);
     }
 
-    /// @dev Used to check the permissions within the upgradability pattern implementation of OZ
-    function _authorizeUpgrade(address)
-        internal
-        virtual
-        override
-        auth(address(this), UPGRADE_ROLE)
-    {}
-
     /// @inheritdoc IPluginRepo
-    function newVersion(
+    function createVersion(
         uint16[3] memory _newSemanticVersion,
-        address _pluginFactoryAddress,
+        address _pluginFactory,
         bytes calldata _contentURI
-    ) external auth(address(this), CREATE_VERSION_ROLE) {
-        // check if _pluginFactoryAddress is IPluginFactory
-        if (!Address.isContract(_pluginFactoryAddress)) {
-            revert InvalidContractAddress({pluginFactoryAddress: _pluginFactoryAddress});
+    ) external auth(address(this), CREATE_VERSION_PERMISSION_ID) {
+        // Check if `_pluginFactory` is a `PluginFactoryBase` contract
+        if (!Address.isContract(_pluginFactory)) {
+            revert InvalidContractAddress({invalidContract: _pluginFactory});
         }
 
         try
-            IPluginFactory(_pluginFactoryAddress).supportsInterface(
+            PluginFactoryBase(_pluginFactory).supportsInterface(
                 PluginFactoryIDs.PLUGIN_FACTORY_INTERFACE_ID
             )
         returns (bool result) {
             if (!result) {
-                revert InvalidPluginInterface({pluginFactoryAddress: _pluginFactoryAddress});
+                revert InvalidPluginFactoryInterface({invalidPluginFactory: _pluginFactory});
             }
         } catch {
-            revert InvalidPluginFactoryContract({pluginFactoryAddress: _pluginFactoryAddress});
+            revert InvalidPluginFactoryContract({invalidPluginFactory: _pluginFactory});
         }
 
-        address basePluginAddress = IPluginFactory(_pluginFactoryAddress).getBasePluginAddress();
+        address basePluginAddress = PluginFactoryBase(_pluginFactory).getBasePluginAddress();
```

Comment:
Why PluginFactoryBase? Why isn't `PluginFactory`meaninfgful?

Reply:
Agreed.

Reply:
We append `Base` if the contract is `abstract`

Reply:
Opened a Jira task for https://aragonassociation.atlassian.net/browse/APP-764. Closing this conversation.